client/dashboard/plugins/manage/components/plugin-switcher.tsx

```diff
@@ -20,12 +22,60 @@ export const PluginSwitcher = ( {
 	searchableFields,
 	selectedPluginSlug = '',
 	view,
+	setView,
+	paginationInfo,
 }: {
 	pluginsWithIcon: PluginListRow[];
 	searchableFields: Field< PluginListRow >[];
 	selectedPluginSlug?: string;
 	view: View;
+	setView: Dispatch< SetStateAction< View > >;
+	paginationInfo: { totalItems: number; totalPages: number };
 } ) => {
+	const scrollRef = useRef< HTMLDivElement >( null );
+
+	// Load next page when scrolling near bottom
+	const handleLoadMore = useCallback( () => {
+		if ( ! paginationInfo ) {
+			return;
+		}
+
+		const currentPage = view.page || 1;
+
+		if ( currentPage < paginationInfo.totalPages ) {
```

Comment:
`currentPage` will always be 1, so I think we should:


```suggestion
		if ( paginationInfo.totalPages > 1 ) {
```